### PR TITLE
First python2 issue of 2020

### PR DESCRIPTION
**Pull request by paulbovbel**
Test build here http://tailor.locusbots.io/blue/organizations/jenkins/ci%2Ftailor-distro/detail/0.1.16/120/pipeline

Newly released setuptools (45) explicitly not compatible with python2, so we need to make sure we install an older one in py2 venvs.

### CODE REVIEW

File: catkin_virtualenv/src/dh_virtualenv/deployment.py [hunk 1]
```diff
@@ -204,8 +181,8 @@ def install_dependencies(self):
             # (pbovbel) allow pinning the pip version
             pip_package = 'pip==' + self.pip_version if self.pip_version else 'pip'
             # First, bootstrap pip with a reduced option set (well-supported options)
-            print(self.pip_preinstall_prefix + self.pip_upgrade_args + ['-U', pip_package])
```
Comment (paulbovbel) [inline catkin_virtualenv/src/dh_virtualenv/deployment.py]: stray print

File: catkin_virtualenv/src/dh_virtualenv/deployment.py [hunk 2]
```diff
@@ -173,11 +149,12 @@ def create_virtualenv(self):
...
-        if self.setuptools:
-            virtualenv.append('--setuptools')
+            # py2's virtualenv command will try install latest setuptools. setuptools>=45 not compatible with py2
+            virtualenv.append('--no-setuptools')
+            self.preinstall = ['setuptools<45' + self.preinstall]
 
-        if self.verbose:
-            virtualenv.append('--verbose')
+            if self.verbose:
+                virtualenv.append('--verbose')
```
Comment (paulbovbel) [inline catkin_virtualenv/src/dh_virtualenv/deployment.py]: --verbose not supported by `python -m venv`

File: catkin_virtualenv/src/dh_virtualenv/deployment.py [hunk 3]
```diff
@@ -130,28 +128,6 @@ def __init__(self,
         # (pbovbel) Set pip_upgrade_args here to keep flags like -q, disregard L111 above.
         self.pip_upgrade_args = self.pip_args
 
-    @classmethod
-    def from_options(cls, package, options):
```
Comment (paulbovbel) [inline catkin_virtualenv/src/dh_virtualenv/deployment.py]: unused method

File: catkin_virtualenv/src/dh_virtualenv/deployment.py [hunk 4]
```diff
@@ -87,7 +86,6 @@ def __init__(self,
         self.pip_version = pip_version
         self.extra_virtualenv_arg = extra_virtualenv_arg
         self.verbose = verbose
-        self.setuptools = setuptools
```
Comment (paulbovbel) [inline catkin_virtualenv/src/dh_virtualenv/deployment.py]: this option is not really relevant - either way you end up with setuptools in the virtualenv

File: catkin_virtualenv/scripts/build_venv [hunk 5]
```diff
@@ -89,7 +89,7 @@ if __name__ == '__main__':
         log_file=None,
         builtin_venv=check_module(python_executable, 'venv'),
         builtin_pip=check_module(python_executable, 'pip'),
-        pip_version='19.0.3'  # (pbovbel) known working version
+        pip_version='19.3.1'  # (pbovbel) known working version
```
Comment (paulbovbel) [inline catkin_virtualenv/scripts/build_venv]: may as well try a new pip

**Comment by knorth55** [comment]
This PR solves our problems!
Do you have any plan to release `catkin_virtualenv` with this commit soon?

**Comment by paulbovbel** [comment]
@knorth55, done, thanks for the reminder.

**Comment by pazeshun** [comment]
@paulbovbel Sorry, but do you have plan to release this to kinetic?

**Comment by pazeshun** [comment]
Thank you for releasing!!!

**Pull request merged**